sklearn/feature_extraction/_hashing_fast.pyx

```diff
@@ -68,7 +68,7 @@ def transform(raw_X, Py_ssize_t n_features, dtype,
             h = murmurhash3_bytes_s32(<bytes>f, seed)
 
             array.resize_smart(indices, len(indices) + 1)
-            indices[len(indices) - 1] = abs(h) % n_features
+            indices[len(indices) - 1] = abs(h % n_features)
```

Comment:
I'd rather only change the result when it was wrong. This way there's no backward compat issue.

```suggestion
            if h == - 2147483648:
                # abs(-2**31) is undefined behavior
                indices[len(indices) - 1] = (2147483647 - (n_features - 1)) % n_features
            else:
                indices[len(indices) - 1] = abs(h) % n_features
```